webdriver/tests/bidi/script/call_function/sandbox.py

```diff
@@ -0,0 +1,153 @@
+import pytest
+
+from webdriver.bidi.modules.script import ContextTarget, ScriptEvaluateResultException
+
+from ... import any_int, any_string, recursive_compare
+from .. import any_stack_trace
+
+
+@pytest.mark.asyncio
+async def test_sandbox(bidi_session, new_tab):
+    # Make changes in window
+    await bidi_session.script.call_function(
+        function_declaration="() => { window.foo = 1; }",
+        target=ContextTarget(new_tab["context"]),
+        await_promise=True,
+    )
+
+    # Check that changes are not present in sandbox
+    result_in_sandbox = await bidi_session.script.call_function(
+        function_declaration="() => window.foo",
+        target=ContextTarget(new_tab["context"], "sandbox"),
+        await_promise=True,
+    )
+    assert result_in_sandbox == {"type": "undefined"}
+
+    # Make changes in sandbox
+    await bidi_session.script.call_function(
+        function_declaration="() => { window.bar = 2; }",
+        target=ContextTarget(new_tab["context"], "sandbox"),
+        await_promise=True,
+    )
+
+    # Make sure that changes are present in sandbox
+    result_in_sandbox = await bidi_session.script.call_function(
+        function_declaration="() => window.bar",
+        target=ContextTarget(new_tab["context"], "sandbox"),
+        await_promise=True,
+    )
+    assert result_in_sandbox == {"type": "number", "value": 2}
+
+    # Make sure that changes didn't leak from sandbox
+    result_in_window = await bidi_session.script.call_function(
+        function_declaration="() => window.bar",
+        target=ContextTarget(new_tab["context"]),
+        await_promise=True,
+    )
+    assert result_in_window == {"type": "undefined"}
+
+
+@pytest.mark.asyncio
+async def test_sandbox_with_empty_name(bidi_session, new_tab):
+    await bidi_session.script.call_function(
+        function_declaration="() => window.foo = 'bar'",
+        target=ContextTarget(new_tab["context"], ""),
```

Comment:
It's more a question to the specification, if we should consider the empty string is a valid sandbox name. But at least a comment clarifying this would be nice.

Reply:
Yeah, we were also not sure if it should be valid, but since the specification doesn't say otherwise, we decided to make sure it works with the test.
I've added a comment.

Reply:
And I've created [a spec issue](https://github.com/w3c/webdriver-bidi/issues/258) to discuss it